nvflare/app_opt/tensor_stream/sender.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+
+from nvflare.apis.fl_constant import FLContextKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.streaming import StreamableEngine, StreamContext
+from nvflare.client.config import ExchangeFormat
+from nvflare.fuel.utils.log_utils import get_obj_logger
+
+from .producer import TensorProducer
+from .types import TENSORS_CHANNEL, TensorsMap
+from .utils import get_dxo_from_ctx, get_targets_for_ctx_and_prop_key, get_topic_for_ctx_prop_key
+
+
+class TensorSender:
+    """Handles sending tensors between server and clients."""
+
+    def __init__(
+        self,
+        engine: StreamableEngine,
+        ctx_prop_key: FLContextKey,
+        format: ExchangeFormat,
+        tasks: list[str],
+        channel: str = TENSORS_CHANNEL,
+    ):
+        """Initialize the TensorSender.
+
+        Args:
+            engine (StreamableEngine): The streamable engine to use for streaming.
+            ctx_prop_key (FLContextKey): The context property key to send tensors for.
```

Comment:
suggest ctx_prop_key (str): The context property key representing which tensors to send.